tests/usb_packet.py

```diff
@@ -58,359 +58,548 @@
+            crc5=0xFF,
+            interEventDelay=interEventDelay,
+        )
+    else:
+        return TokenPacket(
+            pid=USB_PID["SOF"],
+            address=address,
+            endpoint=ep,
+            data_valid_count=data_valid_count,
+            interEventDelay=interEventDelay,
+        )
+
+    return sofToken
 
 
+def reflect(val, numBits):
 
-def AppendSetupToken(packets, ep, address, **kwargs):
-    ipg = kwargs.pop('inter_pkt_gap', 500)
-    AppendTokenPacket(packets, 0x2d, ep, ipg, address, **kwargs)
-
-def AppendOutToken(packets, ep, address, **kwargs):
-    ipg = kwargs.pop('inter_pkt_gap', 500) 
-    AppendTokenPacket(packets, 0xe1, ep, ipg, address, **kwargs)
-
-def AppendPingToken(packets, ep, address, **kwargs):
-    ipg = kwargs.pop('inter_pkt_gap', 500) 
-    AppendTokenPacket(packets, 0xb4, ep, ipg, address, **kwargs)
-
-def AppendInToken(packets, ep, address, **kwargs):
-    #357 was min IPG supported on bulk loopback to not nak
-    #lower values mean the loopback NAKs
-    ipg = kwargs.pop('inter_pkt_gap', 10) 
-    AppendTokenPacket(packets, 0x69, ep, ipg, address, **kwargs)
+    valRef = 0
+    for i in range(numBits):
+        valRef <<= 1
 
-def AppendSofToken(packets, framenumber, **kwargs):
-    ipg = kwargs.pop('inter_pkt_gap', 500) 
-    
-    # Override EP and Address 
-    ep = (framenumber >> 7) & 0xf
-    address = (framenumber) & 0x7f
-    AppendTokenPacket(packets, 0xa5, ep, ipg, address, **kwargs)
+        if val & 1:
+            valRef |= 1
 
-def AppendTokenPacket(packets, _pid, ep, ipg, addr=0, **kwargs):
-    
-    data_valid_count = kwargs.pop('data_valid_count', 0)
-    packets.append(TokenPacket( 
-        inter_pkt_gap=ipg, 
-        pid=_pid,
-        address=addr, 
-        endpoint=ep,
-        data_valid_count=data_valid_count))
+        val >>= 1
 
-def reflect(val, numBits):
+    return valRef
 
-    valRef = 0;
-    for i in range(numBits):
-        valRef <<= 1;
-        
-        if (val & 1):
-            valRef |= 1;
-        
-        val >>= 1;
-    
-    return valRef;
 
 def GenCrc16(args):
 
     data = args
-  
-    crc = 0xffff;
-    poly = 0x8005;
 
-    for byte in  data:
-        topBit = 1 << 15;
-        crc ^= reflect(int(byte) & int(0xFF), 8) << 8; 
+    crc = 0xFFFF
+    poly = 0x8005
 
-        for k in range(0,8):
+    for byte in data:
+        topBit = 1 << 15
+        crc ^= reflect(int(byte) & int(0xFF), 8) << 8
+
+        for k in range(0, 8):
             if crc & topBit:
-                crc = (crc << 1) ^ poly;
+                crc = (crc << 1) ^ poly
             else:
-                crc <<= 1;
+                crc <<= 1
+
+    # //crc = crc ^0xffff;
+    crc = reflect(crc, 16)
+    crc = ~crc
+    crc = crc & 0xFFFF
+    # print "CRC: : {0:#x}".format(crc)
+    return crc
 
-    #//crc = crc ^0xffff;
-    crc = reflect(crc,16);
-    crc = ~crc;
-    crc = crc & 0xffff;
-    #print "CRC: : {0:#x}".format(crc)
-    return crc;
```

Comment:
From memory I think the this was a direct translation from the reference in the spec. Doesn't hurt to speed it up since we're now using it more in anger (Note, largest packet size is 1k)

Also note, this was a formatting only change.

Reply:
@DaveAtkinsonXMOS  This is causing some test fails. I've not debugged, will revert for now.